### ISSUE TITLE
the problem of reproduce

### ISSUE THREAD

**Issue by Win-commit** [baaivision/UniVLA#11]
Thank you very much for your work, but I have a few questions when I reproduce this work.
 For data used to finetune the WORLD_MODEL_POSTTRAIN in libero, I follow the instructions showed in this readme, finetune a single model using all libero data.
And for Policy Learning parameters, I followed the original training ssh(only modificated "save_steps" from 2000 to 800) with the original train file:

```WORLD_SIZE=${WORLD_SIZE:-1}
RANK=${RANK:-0}
MASTER_ADDR=${MASTER_ADDR:-127.0.0.1}
MASTER_PORT=${MASTER_PORT:-23456}
NGPUS=8
export TORCH_EXTENSIONS_DIR=/tmp
DATAPATH='/liujinxin/zhy/UniVLA/datasets/processed_data/meta/libero_all_norm_aug.pkl'
ACTION_TOKENIZER_PATH="/liujinxin/zhy/UniVLA/pretrain/fast"
EXP_NAME="UNIVLA_LIBERO_VIDEO_BS192_8k_reproduce"

export PYTHONPATH=$(pwd)

torchrun \
    --nproc_per_node=${NGPUS} \
    --nnodes=1 \
    --node_rank=${RANK} \
    train/train_moe.py \
    --model_name_or_path /liujinxin/zhy/UniVLA/ckpts/UniVLA/WORLD_MODEL_POSTTRAIN \
    --model_config_path /liujinxin/zhy/UniVLA/configs/moe_fast_video.json \
    --deepspeed scripts/sft/zero3_offload.json \
    --output_dir "logs/"${EXP_NAME} \
    --learning_rate 8e-5 \
    --null_prompt_prob 0.15 \
    --weight_decay 0.1 \
    --min_learning_rate 5e-6 \
    --max_grad_norm 5.0 \
    --adam_beta1 0.9 \
    --adam_beta2 0.95 \
    --adam_epsilon 1e-6 \
    --bf16 True \
    --tf32 True \
    --data_path ${DATAPATH} \
    --max_steps 8000 \
    --dataloader_num_workers 12 \
    --lr_scheduler_type "cosine_with_min_lr" \
    --warmup_steps 50 \
    --per_device_train_batch_size 3 \
    --frames 2 \
    --action_frames 10 \
    --max_position_embeddings 3200 \
    --seed 42 \
    --logging_steps 20 \
    --gradient_checkpointing True \
    --gradient_accumulation_steps 8 \
    --save_strategy steps \
    --save_steps 800 \
    --eval_strategy no \
    --apply_loss_on_only_vision False \
    --apply_loss_on_only_action True \
    --actions True \
    --actions_format "fast" \
    --use_gripper True \
    --video_format "interleave" \
    --action_tokenizer_path ${ACTION_TOKENIZER_PATH} \
    --report_to wandb
```
trained on 8H800.
However, the results I have trained are very different from those reported in your paper. And here are the results of the tests, including the UNIVLA_LIBERO_VIDEO_BS192-8K checkpoints that you have provided.

| MODEL | LIBERO-SPATIAL | LIBERO-OBJECT |  LIBERO-GOAL | LIBERO-10 |
|---|---|---|---|---|
|results in paper | 95.4% | 98.8% | 93.6%  | 94.0% |
| UNIVLA_LIBERO_VIDEO_BS192-8K(provided in readme) | 95.4% | 97.0% | 89.8% | 90.2% |
| checkpoints-8000(I trained myself) | 74.8% | 74.6% | 72.2%|47.2%|
| checkpoints-6400| 80.8%| 67.6%| 68.4%|54.4%|
| checkpoints-4800|81.0%| 78.2%| 72.0%|54.4%|
| checkpoints-1600|91.8%|84.4%| 77.0%|46.2%|

I found that my trained model is very different from the original result in every task suit, but I don't know what's wrong.

**Comment by Robertwyq** [comment]
@Win-commit 

<img width="2144" height="120" alt="Image" src="https://github.com/user-attachments/assets/71f0e8cb-8a52-4aa0-9a48-debddbe4331a" />
I’ve realized there may be two possible issues:
1. Augmentation in the script
I suspect I mistakenly left the aug function in the training script for testing purposes—random_shift isn’t actually needed here. Could you confirm whether you trained using only the augmented data? If that’s the case, it could explain the abnormal point count. The model’s performance should be based on training with normal data.
2. Inference normalization parameters
Please double-check that the hard-coded normalization statistics I included match the values you used during preprocessing. Even small mismatches can lead to performance degradation during inference.

**Comment by Win-commit** [comment]
> [@Win-commit](https://github.com/Win-commit)
> 
> <img alt="Image" width="2000" height="120" src="https://private-user-images.githubusercontent.com/39329846/467875922-71f0e8cb-8a52-4aa0-9a48-debddbe4331a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI4MTgwNTcsIm5iZiI6MTc1MjgxNzc1NywicGF0aCI6Ii8zOTMyOTg0Ni80Njc4NzU5MjItNzFmMGU4Y2ItOGE1Mi00YWEwLTlhNDgtZGViZGRiZTQzMzFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzE4VDA1NDkxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM1MTVlZjY5NTUyNTE2OGZjNGJkMjIwYzc4ZjZiZGVlZmFmMDI4YmY3ODJhZWQ3ZmJjNmM0MmE3NjE1OWJjYmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ELLCQMBpc5T2iW8aPTZd88PwZPP5xbfX9qxp37Qi3zk"> I’ve realized there may be two possible issues: 1. Augmentation in the script I suspect I mistakenly left the aug function in the training script for testing purposes—random_shift isn’t actually needed here. Could you confirm whether you trained using only the augmented data? If that’s the case, it could explain the abnormal point count. The model’s performance should be based on training with normal data. 2. Inference normalization parameters Please double-check that the hard-coded normalization statistics I included match the values you used during preprocessing. Even small mismatches can lead to performance degradation during inference.

For the first reason: I didn't modify the code in **emu3_tokenizer.py**, so the parameter 'augmentation' is random_shift, not None. should the correct version  be 
`images, image_paths = load_images(osp.join(VIDEO_ROOT, video, 'images'), SIZE, interval)` ?
For the second reason,where can I find the hard-coded normalization statistics? Is it in `UniVLA/reference/RoboVLMs/eval/libero/model_wrapper_emu.py`, the **unormalize_action** function? I didn't modify this part.
So I guess maybe the random_shift augmentation causes the error distribution in dataset, I would train it again with the correct data. Thanks for your answering

**Comment by Robertwyq** [comment]
@Win-commit 

It seems like your model’s performance has deteriorated over time—likely because it’s overfitting to the augmented data distribution. Apologies — when I uploaded the code, I forgot to remove the augmentation step. Your fix is correct!
Also remember to process the gripper image.

Secondly, there’s also an issue in the unnormalize_action function that needs attention.

**Comment by Win-commit** [comment]
> > [@Win-commit](https://github.com/Win-commit)
> > <img alt="Image" width="2000" height="120" src="https://private-user-images.githubusercontent.com/39329846/467875922-71f0e8cb-8a52-4aa0-9a48-debddbe4331a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI4MTgwNTcsIm5iZiI6MTc1MjgxNzc1NywicGF0aCI6Ii8zOTMyOTg0Ni80Njc4NzU5MjItNzFmMGU4Y2ItOGE1Mi00YWEwLTlhNDgtZGViZGRiZTQzMzFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzE4VDA1NDkxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM1MTVlZjY5NTUyNTE2OGZjNGJkMjIwYzc4ZjZiZGVlZmFmMDI4YmY3ODJhZWQ3ZmJjNmM0MmE3NjE1OWJjYmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ELLCQMBpc5T2iW8aPTZd88PwZPP5xbfX9qxp37Qi3zk"> I’ve realized there may be two possible issues: 1. Augmentation in the script I suspect I mistakenly left the aug function in the training script for testing purposes—random_shift isn’t actually needed here. Could you confirm whether you trained using only the augmented data? If that’s the case, it could explain the abnormal point count. The model’s performance should be based on training with normal data. 2. Inference normalization parameters Please double-check that the hard-coded normalization statistics I included match the values you used during preprocessing. Even small mismatches can lead to performance degradation during inference.
> 
> For the first reason: I didn't modify the code in **emu3_tokenizer.py**, so the parameter 'augmentation' is random_shift, not None. should the correct version be `images, image_paths = load_images(osp.join(VIDEO_ROOT, video, 'images'), SIZE, interval)` ? For the second reason,where can I find the hard-coded normalization statistics? Is it in `UniVLA/reference/RoboVLMs/eval/libero/model_wrapper_emu.py`, the **unormalize_action** function? I didn't modify this part. So I guess maybe the random_shift augmentation causes the error distribution in dataset, I would train it again with the correct data. Thanks for your answering

I have resolved this problem, the reason is actually the **random_shift** arguement,and the performence now is the same as the report
